contracts/src/ocr2/aggregator.cairo

```diff
@@ -292,20 +313,23 @@ mod Aggregator {
         _billing: BillingConfig,
         // payee management
         _payees: LegacyMap<ContractAddress, ContractAddress>, // <transmitter, payment_address>
-        _proposed_payees: LegacyMap<ContractAddress,
-        ContractAddress> // <transmitter, payment_address>
+        _proposed_payees: LegacyMap<
+            ContractAddress, ContractAddress
+        > // <transmitter, payment_address>
     }
 
-    fn _require_read_access() {
-        let caller = starknet::info::get_caller_address();
-        let state = AccessControl::unsafe_new_contract_state();
-        AccessControl::check_read_access(@state, caller);
+    #[generate_trait]
+    impl AccessHelperImpl of AccessHelperTrait {
+        fn _require_read_access(self: @ContractState) {
+            let caller = starknet::info::get_caller_address();
+            self.access_control.check_read_access(caller);
+        }
     }
 
     #[external(v0)]
```

Comment:
shouldn't this be `#[abi(embed_v0)]`? Goes for all the other contracts as well (ref: https://book.cairo-lang.org/ch99-01-03-02-contract-functions.html#2-public-functions)

Reply:
good catch, `external(v0)` is being deprecated but seems to still work right now

Reply:
looks like deprecation warning is added in 2.5.0